example/cajita_tutorial/07_local_mesh/local_mesh_example.cpp

```diff
@@ -0,0 +1,141 @@
+
+    using exec_space = Kokkos::DefaultHostExecutionSpace;
+    using device_type = exec_space::device_type;
+
+    /*
+      As with the previous examples, we first create the global mesh and grid,
+      as well as the local grid.
+    */
+    std::array<double, 3> low_corner = { -1.2, 0.1, 1.1 };
+    std::array<double, 3> high_corner = { -0.3, 9.5, 2.3 };
+    double cell_size = 0.05;
+    auto global_mesh =
+        Cajita::createUniformGlobalMesh( low_corner, high_corner, cell_size );
+
+    // Here we partition only in x to simplify the example below.
+    int comm_size;
+    MPI_Comm_size( MPI_COMM_WORLD, &comm_size );
+    std::array<int, 3> input_ranks_per_dim = { comm_size, 1, 1 };
+    Cajita::ManualBlockPartitioner<3> partitioner( input_ranks_per_dim );
+
+    // Create the global grid.
+    std::array<bool, 3> is_dim_periodic = { true, true, true };
+    auto global_grid = Cajita::createGlobalGrid( MPI_COMM_WORLD, global_mesh,
+                                                 is_dim_periodic, partitioner );
+
+    // Create a local grid
+    int halo_width = 1;
+    auto local_grid = Cajita::createLocalGrid( global_grid, halo_width );
+
+    // Create the local mesh from the local grid.
+    auto local_mesh = Cajita::createLocalMesh<device_type>( *local_grid );
+
+    /*
+      Just like the global mesh, the local mesh holds information about the
+      physical exents and dimensions of the cells. The local mesh is distinct in
+      that it is intended to be regularly used at the application level and is
+      designed to be used on the host or in device kernels.
+    */
+    std::cout << "Low corner local: ";
+    for ( int d = 0; d < 3; ++d )
+        std::cout << local_mesh.lowCorner( Cajita::Own(), d ) << " ";
+    std::cout << "\nHigh corner local: ";
+    for ( int d = 0; d < 3; ++d )
+        std::cout << local_mesh.highCorner( Cajita::Own(), d ) << " ";
+    std::cout << "\nExtent local: ";
+    for ( int d = 0; d < 3; ++d )
+        std::cout << local_mesh.extent( Cajita::Own(), d ) << " ";
+    std::cout << "\nLow corner ghost: ";
+    for ( int d = 0; d < 3; ++d )
+        std::cout << local_mesh.lowCorner( Cajita::Ghost(), d ) << " ";
+    std::cout << "\nHigh corner ghost: ";
+    for ( int d = 0; d < 3; ++d )
+        std::cout << local_mesh.highCorner( Cajita::Ghost(), d ) << " ";
+    std::cout << "\nExtent ghost: ";
+    for ( int d = 0; d < 3; ++d )
+        std::cout << local_mesh.extent( Cajita::Ghost(), d ) << " ";
+
+    /*
+      Note that this information is taken directly from the global grid and mesh
+      information.
+    */
+    std::cout << "\n\nLow corner from global offset ";
+    for ( int d = 0; d < 3; ++d )
+        std::cout << cell_size * global_grid->globalOffset( d ) << " ";
+
+    /*
+      The local mesh is most often useful to get information about individual
+      cells within a kernel. Note this is done on the host here, but would most
+      often be in a Kokkos parallel kernel.
```

Comment:
I would point out that the local mesh object is designed such that it can be captured by value in a Kokkos parallel loop